src/k8s/pkg/k8sd/app/hooks_remove.go

```diff
@@ -145,5 +146,65 @@ func (a *App) onPreRemove(ctx context.Context, s state.State, force bool) (rerr
 		}
 	}
 
+	tryCleanupContainerdPaths(snap)
+
 	return nil
 }
+
+// tryCleanupContainerdPaths attempts to clean up all containerd directories which were
+// created by the k8s-snap based on the existence of their respective lockfiles
+// located in the directory returned by `s.LockFilesDir()`.
+func tryCleanupContainerdPaths(s snap.Snap) {
+	log := log.L()
```

Comment:
One last thing before merging, can we pass the `onPreRemove` logger through the context and use it here?